backend/dataall/modules/shares_base/db/share_object_models.py

```diff
@@ -57,3 +58,17 @@ class ShareObjectItem(Base):
     healthStatus = Column(String, nullable=True)
     healthMessage = Column(String, nullable=True)
     lastVerificationTime = Column(DateTime, nullable=True)
+    attachedDataFilterUri = Column(
```

Comment:
We must be aware that this field is now affecting any type of share object (soon redshift shares, but also future notebook shares, X shares...) At first it does not seem to be very S3-Dataset-LF-filter specific, so I am okey introducing data filters to the share object; I'll review that there are no S3-LF specific assumptions in the shares_base module, but I wanted to bring it up because you know the code better

Reply:
yeah I made data filter a generic `share_base` component thinking that the concept of share data filters is not necessarily table specific - only parts of the implementation are specific to handling table share items and LF

Think about for data filters maybe introducing a regex pattern to exclude could be an option on bucket or table shares and we could reuse this concept of attached filters?  Open to feedback if you think about it differently